### PR TITLE
[WIP] assert that settings are not empty

### PR DESCRIPTION
**Pull request by CheSema**
Related to https://github.com/ClickHouse/ClickHouse/issues/44271

WriteBufferFromS3 could run wrong if settings are empty.
Firstly determine the sensitive set of tests, see what will fail with assertions.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by tavplubix** [comment]
Does it make sense to remove default values because these values are incorrect according to the assertions?
https://github.com/ClickHouse/ClickHouse/blob/b91af1b650c90ad8b5065a6af9f90945d8d7c23b/src/Storages/StorageS3Settings.h#L29-L39

**Comment by CheSema** [comment]
> Does it make sense to remove default values because these values are incorrect according to the assertions?

That 0-values is incorrect for following logic, if they are not set that leads to disaster. As I understand, the intension to do such initiation is to determine lates is settings are set or not, therefore this is the way to handle the case when some settings should be overridden from other source of settings, overridden happens only if the settings is 0 (not set).
Usually to face that is better to use std::optional<> var.

In short, yes, default values have to be correct. I intend to change this part.

The review isn't final, I just try to see what tests are sensitive and take them as an example for new test.

**Comment by KochetovNicolai** [comment]
https://github.com/ClickHouse/ClickHouse/pull/44335
